airbyte/_executors/util.py

```diff
@@ -242,6 +246,8 @@ def get_connector_executor(  # noqa: PLR0912, PLR0913, PLR0914, PLR0915, C901 #
                 case InstallType.PYTHON:
                     pip_url = metadata.pypi_package_name
                     pip_url = f"{pip_url}=={version}" if version else pip_url
+                case InstallType.JAVA:
+                    docker_image = True
```

Comment:
Let's make this a bit more robust...

When default install type is Java:
   1. If `use_java_tar` is False: fallback to Docker.
   2. If `use_java_tar` is None and Docker is available: use Docker.
   3. Else (`use_java_tar` is Truthy): use Java.